toapi/selector.py

```diff
@@ -34,7 +34,7 @@ def parse(self, html):
             # TODO
             value = value[0].get(self.attr).strip() if len(value) == 1 else value
         else:
-            if len(value) == 1:
+            if isinstance(value, list) and len(value) == 1 and isinstance(value[0], etree._Element):
```

Comment:
Your solution makes thing complex. I just want to get the first item of the XPath result list. Css selector still not works well.

Reply:
I have already sent an email to give you an explanation.